lib/dry/schema/path.rb

```diff
@@ -90,15 +90,29 @@ def index(key)
         keys.index(key)
       end
 
+      def without_index
+        self.class.new([*to_a[0..-2]])
+      end
+
       # @api private
+      #
+      # rubocop:disable Metrics/PerceivedComplexity
       def include?(other)
         return false unless same_root?(other)
         return last.equal?(other.last) if index? && other.index?
-        return self.class.new([*to_a[0..-2]]).include?(other) if index?
-        return false if !index? && other.index?
+        return without_index.include?(other) if index?
+
+        if !index? && other.index?
+          path = key_matches(other, :select)
+
+          return false unless path.size > 1
+
+          self.class.new(path).include?(other)
+        end
```

Comment:
Equivalent to this code
```ruby
      def include?(other)
        return false unless same_root?(other)
        return last.equal?(other.last) if index? && other.index?
        return without_index.include?(other) if index?
        return false if other.index? && key_matches(other, :select).size < 2

        self >= other && !other.key_matches(self).include?(nil)
      end
```
But I would prefer a form without `return`:
```ruby
      def include?(other)
        if !same_root?(other)
          false
        elsif index?
          if other.index?
            last.equal?(other.last)
          else
            without_index.include?(other)
          end
        elsif other.index? && key_matches(other, :select).size < 2
          false
        else
          self >= other && !other.key_matches(self).include?(nil)
        end
      end
```
because the latter doesn't have duplicated checks (see calls of `index?`) and IMHO a bit less prone to error